test/server/admin/grouped_stats_request_test.cc

```diff
@@ -1,62 +1,63 @@
-#include <regex>
-#include <string>
-#include <vector>
+#include <memory>
```

Comment:
Refactoring/rewriting all the tests makes it EXTREMELY difficult to understand if there were any behavior changes to what prometheus stats were output. Can you try leave the prometheus tests as untouched as possible in this PR, and we can consider revising them later?

Also, I think it makes sense to leave a whole set of prometheus-format-specific tests.

Reply:
Fair point about the difficulty of reviewing due to moving all the tests. I will move the pre-existing prom related ones back to original files, hopefully that will ease code review.
 
> Also, I think it makes sense to leave a whole set of prometheus-format-specific tests.
Yeah, all of the pre-existing prometheus tests are left pretty much untouched, I only added new tests in this patch. Or maybe I am misunderstanding what you mean with this comment?